Python-path-traversal-files-main/747929791/majsoul_wrapper/sdk.py

```
@@ -0,0 +1,483 @@
+
+def dumpWebSocket(filename='ws_dump.pkl'):
+    # 监听mitmproxy当前websocket，将所有报文按顺序交由handler.parse
+    server = ServerProxy("http://127.0.0.1:37247")  # 初始化服务器
+    liqi = LiqiProto()
+    tot = 0
+    history_msg = []
+    handler = MajsoulHandler()
+    while True:
+        n = server.get_len()
+        if tot < n:
+            flow = pickle.loads(server.get_items(tot, n).data)
```

Ruff (0.13.3)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)